server/src/main/java/io/crate/fdw/ForeignTableRelation.java

```diff
@@ -57,7 +60,17 @@ public Symbol getField(ColumnIdent column,
         }
         Reference reference = tableInfo.getReadReference(column);
         if (reference == null) {
-            throw new ColumnUnknownException(column, tableInfo.name());
+            // Try to get the root of the column e.g. for json_col['x']['y'], get `json_col`
+            reference = tableInfo.getReadReference(ColumnIdent.of(column.name()));
+            if (reference == null) {
+                throw new ColumnUnknownException(column, tableInfo.name());
+            }
+            reference = new SimpleReference(
+                new ReferenceIdent(relationName(), column),
+                reference.granularity(),
+                DataTypes.STRING,
+                reference.position(),
+                reference.defaultExpression());
         }
```

Comment:
Looks okay to me to go for a top column. It's correlates with `getField` javadocs in the interface.

>They can return `Reference` symbols for subscripts; In that case on the `topLevel` part of the column appears in the `output`

Comparing with `DocTableRelation` implementation, I found 2 discrepancies, wonder if we should align with it here:

1. Reference being not found is not enough to throw NPE. It's controlled by `errorOnUnknownObjectKey`.
2. We use `DynamicReference`, not SimpleReference.

In general, is it possible to re-use `DocTableInfo.getDynamic` here if `Reference` is null? Maybe extracting some common parts if not applicable as is.